src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -107,11 +107,20 @@ uct_cuda_ipc_iface_is_reachable_v2(const uct_iface_h tl_iface,
     uct_cuda_ipc_md_t *md        = ucs_derived_of(base_iface->md, uct_cuda_ipc_md_t);
 #endif
 
-    if (!uct_iface_is_reachable_params_addrs_valid(params) ||
-        (getpid() == *(pid_t*)params->iface_addr) ||
-        !uct_iface_scope_is_reachable(tl_iface, params)) {
+    if (!uct_iface_is_reachable_params_addrs_valid(params)) {
         return 0;
-    };
+    }
+
+    if (getpid() == *(pid_t*)params->iface_addr) {
+        uct_iface_fill_info_str_buf(params,
+                                    "different pid %d vs %d", getpid(),
```

Comment:
"same process"
since cuda_ipc cannot connect within the same process - check is == not !=